### DISCUSSION TITLE
Are types planned?

### DISCUSSION THREAD

**Discussion by alita-moore** [statelyai/xstate#4575] [General]
I noticed in v5 there's no great way to type the return types and input variables into actors from promises. I'm just getting used to it now so maybe I'm missing something. Is a more robust type support there on the roadmap? Or is it not possible?

By the way, I mean typing in the machine definition and in the .provide method

**Comment by alita-moore** [comment]
nvm I think I see how to do it using the types attribute, that's pretty neat!

**Reply by davidkpiano** [reply]
Quick reference for others: https://stately.ai/docs/cheatsheet#types
